src/meta_schedule/postproc/rewrite_unbound_block.cc

```diff
@@ -195,17 +200,31 @@ bool RewriteUnboundBlockNode::Apply(const tir::Schedule& sch) {
     if (bind_type == tir::BindType::kBindBlock) {
       sch->Bind(fused, "blockIdx.x");
     } else if (bind_type == tir::BindType::kBindBlockThread) {
-      Array<LoopRV> splits = sch->Split(fused, {NullOpt, Integer(this->warp_size_)});
-      ICHECK_EQ(splits.size(), 2);
-      sch->Bind(splits[0], "blockIdx.x");
-      sch->Bind(splits[1], "threadIdx.x");
+      int extent_size = 0, max_block = 256;
+      if (const IntImmNode* node = sch->Get(fused)->extent.as<IntImmNode>()) {
+        extent_size = node->value;
+      }
+      Array<LoopRV> splits;
+      if (extent_size > max_block * max_num_threads_) {
+        splits = sch->Split(fused, {NullOpt, Integer(max_block), Integer(max_num_threads_)});
+        ICHECK_EQ(splits.size(), 3);
+        sch->Reorder({splits[1], splits[2], splits[0]});
+        sch->Bind(splits[1], "blockIdx.x");
+        sch->Bind(splits[2], "threadIdx.x");
+      } else if (extent_size != 0) {
+        splits = sch->Split(fused, {NullOpt, Integer(std::min(max_num_threads_, extent_size))});
+        ICHECK_EQ(splits.size(), 2);
+        sch->Bind(splits[0], "blockIdx.x");
+        sch->Bind(splits[1], "threadIdx.x");
+      }
     }
   }
   return true;
 }
 
 Postproc Postproc::RewriteUnboundBlock() {
   ObjectPtr<RewriteUnboundBlockNode> n = make_object<RewriteUnboundBlockNode>();
+  n->max_num_threads_ = -1;
   n->warp_size_ = -1;
```

Comment:
Could you confirm if `warp_size` is used any more? If not, let's simply remove this field